### PR TITLE
fix: functioning disclaimer dialog

### PR DESCRIPTION
**Pull request by ColinBuyck**
# Pull Request Template

## Issue Overview

This PR addresses #470 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR covers the frontend needs of the export disclaimer feature. There is still some outstanding work that will happen in Core around updating the logger so that exports by user will be tracked on the db.

Since this mirrors the needs of the lottery disclaimer, I made a shared component to handle both cases when the lottery feature is ready.

Note: that I've observed some inconsistent behavior on the applications export email but it is only temporarily an email until prisma is brought over to Doorway and the onExport seems to be called correctly

## How Can This Be Tested/Reviewed?

This can be tested by going to the applications page and attempting to export a listing.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/src/pages/listings/[id]/applications/index.tsx [hunk 1]
```diff
@@ -189,14 +201,39 @@ const ApplicationsList = () => {
...
+                  {t("applications.export.termsOfUse")}
+                </h2>
+                <span>
+                  {t("applications.export.termsBodyOne")}
+                  <a href="https://mtc.ca.gov/doorway-housing-portal-terms-use">
+                    {t("applications.export.termsOfUse")}
+                  </a>
+                  {t("applications.export.termsBodyTwo")}
+                  <a href="https://docs.google.com/document/d/1W4tIMtUMwz4KqdcO5f4yZi0R5AU74P3B/edit?usp=sharing&ouid=105961542504967611474&rtpof=true&sd=true">
+                    {t("applications.export.doorwayPartnersManual")}
```
Comment (emilyjablonski) [inline sites/partners/src/pages/listings/[id]/applications/index.tsx:229]: We shouldn't be concatenating strings for translations, could we include the URL in the string and pass in the string and URL as vars?

Comment (ColinBuyck) [reply sites/partners/src/pages/listings/[id]/applications/index.tsx:229]: Per oflfine conversation, I've moved these strings to the partners json files @emilyjablonski

File: sites/partners/src/pages/listings/[id]/applications/index.tsx [hunk 2]
```diff
@@ -189,14 +201,39 @@ const ApplicationsList = () => {
...
+                onClose={() => setIsTermsOpen(false)}
+                onSubmit={onSubmit}
+              >
+                <p>{t("applications.export.dialogSubheader")}</p>
+                <h2 className={styles["terms-of-use-text"]}>
+                  {t("applications.export.termsOfUse")}
+                </h2>
+                <span>
+                  {t("applications.export.termsBodyOne")}
+                  <a href="https://mtc.ca.gov/doorway-housing-portal-terms-use">
```
Comment (emilyjablonski) [inline sites/partners/src/pages/listings/[id]/applications/index.tsx:224]: Could we open these in a new tab?

**Review by emilyjablonski — APPROVED** [review]
Screenshot 2024-03-12 at 12 25 41 PM
Non-blocking nit that the padding isn't consistent around the checkbox

**Comment by ColinBuyck** [comment]
Fixed @emilyjablonski! Good catches 🥼

**Pull request merged**